### PR TITLE
refactor unit tests for namespace and operatorGroup

### LINKED ISSUE #243: Refactor namespace and operator group unit tests
refactor existing `namespace_tests.go` and `operator_group_tests.go` to split them into multiple unit tests instead of just one function.

### PR DESCRIPTION
**Pull request by yashoza19**
## Description of PR
Fixes #243.

## Changes (required)

- namespace_test.go
- operator_group_test.go

## Checklist (required)
- [x] I have reviewed and followed the [contribution guidelines](https://github.com/opdev/opcap/blob/main/docs/contribution.md)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if needed
- [x] I have checked that my changes pass all tests

Signed-off-by: yoza <yoza@redhat.com>

### CODE REVIEW

**Review by acmenezes — APPROVED** [review]
That looks great!

File: internal/operator/namespace_test.go [hunk 1]
```diff
@@ -19,23 +19,40 @@ var _ = Describe("Namespace", func() {
 	var operatorClient operatorClient = operatorClient{
 		Client: client,
 	}
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:21]: This code needs to be in a BeforeEach()

File: internal/operator/namespace_test.go [hunk 2]
```diff
@@ -19,23 +19,40 @@ var _ = Describe("Namespace", func() {
...
-	Context("Namespaces", func() {
-		It("should exercise Namespaces", func() {
-			By("creating a Namespace", func() {
-				ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
+
+	Context("CreateNamespace", func() {
+		When("creating a namespace", func() {
+			ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:25]: This should be inside the It()

File: internal/operator/namespace_test.go [hunk 3]
```diff
@@ -19,23 +19,40 @@ var _ = Describe("Namespace", func() {
...
 			})
-			By("creating it again should error", func() {
+		})
+		When("creating a namespace that already exists", func() {
+			JustBeforeEach(func() {
 				ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
-				Expect(err).To(HaveOccurred())
-				Expect(err).To(MatchError("could not create Namespace: testns: namespaces \"testns\" already exists"))
-				Expect(ns).To(BeNil())
+				It("should error", func() {
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:34]: This should not be inside the JustBeforeEach

File: internal/operator/namespace_test.go [hunk 4]
```diff
@@ -19,23 +19,40 @@ var _ = Describe("Namespace", func() {
...
+			ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
+			It("should succeed", func() {
 				Expect(err).ToNot(HaveOccurred())
 				Expect(ns).ToNot(BeNil())
 			})
-			By("creating it again should error", func() {
+		})
+		When("creating a namespace that already exists", func() {
+			JustBeforeEach(func() {
 				ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:33]: This line should be in both the JustBeforeEach, and in the It()

File: internal/operator/namespace_test.go [hunk 5]
```diff
@@ -19,23 +19,40 @@ var _ = Describe("Namespace", func() {
...
+					Expect(ns).To(BeNil())
+				})
 			})
-			By("deleting that Namespace", func() {
-				err := operatorClient.DeleteNamespace(context.TODO(), "testns")
+		})
+	})
+	Context("DeleteNamespace", func() {
+		When("deleting the existing namespace", func() {
+			err := operatorClient.DeleteNamespace(context.TODO(), "testns")
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:44]: This should be inside the It()

File: internal/operator/namespace_test.go [hunk 6]
```diff
@@ -19,23 +19,40 @@ var _ = Describe("Namespace", func() {
...
+			It("should succeed", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
+		When("deleting a namespace that does not exist", func() {
+			JustBeforeEach(func() {
+				err := operatorClient.DeleteNamespace(context.TODO(), "testns")
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:51]: Same as the Create

**Review by skattoju — APPROVED** [review]
/lgtm

File: internal/operator/operator_group_test.go [hunk 7]
```diff
@@ -19,28 +19,44 @@ var _ = Describe("OperatorGroup", func() {
 	var operatorClient operatorClient = operatorClient{
 		Client: client,
 	}
+	operatorGroupData := OperatorGroupData{
```
Comment (bcrochet) [inline internal/operator/operator_group_test.go:22]: Generally same comments as NameSpace.

**Review by bcrochet — CHANGES_REQUESTED** [review]
🛑  A few changes

**Pull request merged**